src/bundles/files.js

```diff
@@ -361,9 +350,9 @@ export default (opts = {}) => {
         content: pageContent.content.sort((a, b) => {
           if (a.type === b.type || isMac) {
             if (sorting.by === sorts.BY_NAME) {
-              return compare(a.name, b.name, sorting.asc)
+              return sortByName(sorting.asc ? 1 : -1)(a.name, b.name)
```

Comment:
the outer function is a comparartor  as well, so it gets called once for every comparison. It'd be reasonanble to construct these comparators ouside the function.

Reply:
Is [this](https://github.com/ipfs-shipyard/ipfs-webui/pull/977/commits/87d775027cefa4bc937e552709f92248f2137b41) what you mean?

If you were referring to the `sorting.by`,

```js
const sortDir = sorting.asc ? 1 : -1
const sortByFn = sorting.by === sorts.BY_NAME ? sortByName : sortBySize

return {
	...pageContent,
	content: pageContent.content.sort((a, b) => {
		if (a.type === b.type || isMac) {
			// the sorting fns have diff params
			// sortByName(sortDir)(a.name, b.name)
			// sortBySize(sortDir)(a.cumulativeSize || a.size, b.cumulativeSize || b.size)
			sortByFn(sortDir)(...)
		}
	})
}
```

Reply:
not quite. we know the direction before we get to `pageContent.content.sort` so we can build the `sortBySize(sortDir)` and `sortByName(sortDir)` comparators before hand.

Reply:
Sorry, but I'm not sure I follow 🤔

Reply:
calling `sortByName(dir)` creates and returns a function, We can create that function outside of the comparator, so we dont re-create it once for every pair of items. we just create it once before we start the sort.

Reply:
```js
const nameSort = sortByName(dir)

//...

return {
	...pageContent,
	content: pageContent.content.sort((a, b) => {
		if (a.type === b.type || isMac) {
		     nameSort(a.name, b.name)
  // ...
		}
	})
}

Reply:
<img src="https://media.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif" />